### PR TITLE
CAB-4053 (part 1): Adds capability to SearchResultsComponent to render in "FreezeResults" mode.

### PR DESCRIPTION
**Pull request by farmas**

### CODE REVIEW

File: src/app/search/shared/model/search-datasource.model.ts [hunk 1]
```diff
@@ -13,9 +13,7 @@ export class SearchDataSource extends DataSource<ResultRow> {
   constructor(
     private searchModel$: Observable<SearchModel>,
     private searchResults$: Observable<SearchResults>,
-    private sort: MatSort,
-    private paginators: Array<MatPaginator>
```
Comment (farmas) [inline src/app/search/shared/model/search-datasource.model.ts:17]: Removed parameter not used anywhere.

File: src/app/search/search-results/search-results.component.spec.ts [hunk 2]
```diff
@@ -7,46 +7,21 @@ import { SearchModel } from '../shared/model/search-model';
...
-import { ActivatedRouteStub } from '../../../testing/router-stubs';
-
-class MockDataService {
-  storage = ['123', '456'];
-}
-
-class RouterStub {
-  navigate(url: string) {
-    return url;
-  }
```
Comment (farmas) [inline src/app/search/search-results/search-results.component.spec.ts:26]: Removed all these mocks in favor of RouterTestingModule.

File: src/app/search/search-results/search-results.component.ts [hunk 3]
```diff
@@ -43,6 +43,8 @@ export class SearchResultsComponent implements OnInit, OnDestroy {
   searchModel$: Observable<SearchModel>;
   @Input()
   searchResults$: Subject<SearchResults>;
+  @Input()
+  readOnly = false;
```
Comment (farmas) [inline src/app/search/search-results/search-results.component.ts:47]: I admit that 'readOnly' might not be the best name for this input, I am open to suggestions.

File: src/app/search/search-results/search-results.component.html [hunk 4]
```diff
@@ -1,5 +1,6 @@
 <div class="search-results-component">
   <mat-paginator #topPaginator
+                 *ngIf="!readOnly"
```
Comment (agagne) [inline src/app/search/search-results/search-results.component.html:3]: Why is pagination disabled for `readOnly`?

Comment (agagne) [reply src/app/search/search-results/search-results.component.html:3]: Nevermind, I think I understand.. when it is not 'readOnly' that means the search-results are in the 'bulk-edit' mode.

File: src/app/search/search-results/search-results.component.ts (hunk 3 above)
Comment (agagne) [reply src/app/search/search-results/search-results.component.ts:47]: Would you provide a comment that explains that readOnly is false, that the search-results are selectable for bulk-edit.

I'm not sure what the right wording is either, readOnly seems wrong.. theoretically I could see this for other uses such as bulk download, viewing multiple documents at once, etc. Maybe something like 'selectable' but that also doesn't seem like the right choice.

File: src/app/search/search-results/search-results.component.html (hunk 4 above)
Comment (farmas) [reply src/app/search/search-results/search-results.component.html:3]: I think that is more evidence that 'readOnly' is not the right term. The intent is to kind of 'freeze' the table, meaning, a user can no longer paginate, sort or use any of the link navigation. This is used in two places: 1) in the search page, when the user selects which rows to include in the bulk edit operation and 2) in the bulk edit page, to just display the rows that are going to be updated.

Maybe call the property 'freezeNavigation' ??

File: src/app/search/search-results/search-results.component.ts (hunk 3 above)
Comment (farmas) [reply src/app/search/search-results/search-results.component.ts:47]: 'selectable' will be a separate toggle, the reason is that in the search page we need the table to be readOnly and selectable, but in the bulk edit page we need the table to be readonly and non-selectable.

File: src/app/search/search-results/search-results.component.html (hunk 4 above)
Comment (agagne) [reply src/app/search/search-results/search-results.component.html:3]: 'freezeNavigation' or 'freezeResults' makes sense

**Pull request merged**